<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Training materials - RSK's test doc</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="../css/highlight.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Training materials";
    var mkdocs_page_input_path = "README.md";
    var mkdocs_page_url = "/README/";
  </script>
  
  <script src="../js/jquery-2.1.1.min.js"></script>
  <script src="../js/modernizr-2.8.3.min.js"></script>
  <script type="text/javascript" src="../js/highlight.pack.js"></script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> RSK's test doc</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1 current">
		
    <a class="current" href="./">Training materials</a>
    <ul class="subnav">
            
    <li class="toctree-l2"><a href="#training-program-description">Training program description:</a></li>
    
        <ul>
        
            <li><a class="toctree-l3" href="#workshops">Workshops</a></li>
        
            <li><a class="toctree-l3" href="#in-depth-next-generation-sequencing-analysis-course">In-Depth Next Generation Sequencing Analysis Course</a></li>
        
            <li><a class="toctree-l3" href="#contact-us">Contact us:</a></li>
        
        </ul>
    

    </ul>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">RSK's test doc</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Training materials</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="training-program-description">Training program description:</h1>
<p>The training team at the Harvard Chan Bioinformatics Core provides bioinformatics training through both shorter workshops and longer in-depth courses. Our current workshops and courses are designed to help biologists become comfortable with using tools to analyse high-throughput data. We are slowly beginning to expand this repertoire to include training for researchers with more advanced bioinformatics skills.</p>
<p><em>No prior NGS or command line expertise is required for our workshops or courses unless explicitly stated.</em></p>
<h3 id="workshops">Workshops</h3>
<h4 id="introduction-to-next-generation-sequencing-ngs-analysis-series">Introduction to Next-Generation Sequencing (NGS) analysis series:</h4>
<p>The goal of these workshops (2-3 days) are to enable researchers to design their NGS studies appropriately and perform preliminary data analyses.</p>
<table>
<thead>
<tr>
<th align="center">Training topic and link to lessons</th>
<th align="center">Prerequisites</th>
<th align="center">Workshop Duration</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Intro-to-R/">R and ggplot2</a></td>
<td align="center">None</td>
<td align="center">2 days</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/">RNA-seq data analysis using High-Performance Computing</a></td>
<td align="center">None</td>
<td align="center">2 - 3 days</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Intro-to-R-with-DGE/">R and Differential Gene Expression (DGE) analysis</a></td>
<td align="center">None</td>
<td align="center">3 days</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/DGE_workshop/">Differential Gene Expression (DGE) analysis</a></td>
<td align="center"><a href="https://hbctraining.github.io/Intro-to-R/">R and ggplot2</a></td>
<td align="center">1.5 days</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Intro-to-ChIPseq/">ChIP-seq using High-Performance Computing</a></td>
<td align="center">None</td>
<td align="center">3 days</td>
</tr>
<tr>
<td align="center">Identifying variants in genome/exome sequencing data</td>
<td align="center">None</td>
<td align="center"><em>In development</em></td>
</tr>
</tbody>
</table>
<h4 id="current-topics-in-bioinformatics-series">Current topics in bioinformatics series:</h4>
<p>These short workshops (half-day or less) are designed to allow researchers, who have some familiarity with R or bash, to learn new tools and methods. Only a subset of the short workshops topics are listed and linked below, for the full list please <a href="https://hbctraining.github.io/Training-modules/">click here</a>.</p>
<table>
<thead>
<tr>
<th align="center">Training topic and link to lessons</th>
<th align="center">Prerequisites</th>
<th align="center">Date &amp; Location</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/IntroR_ggplot2/">Introduction to R &amp; Visualizations with ggplot2</a></td>
<td align="center">None</td>
<td align="center">TBD</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Visualization_in_R/">Plotting and visualization in R using ggplot2</a></td>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/IntroR_ggplot2/">Intro to R</a></td>
<td align="center">TBD</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Intro_shell/">Introduction to shell/bash</a></td>
<td align="center">None</td>
<td align="center">TBD</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Intermediate_shell/">Intermediate shell/bash</a></td>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Intro_shell/">Intro to shell</a></td>
<td align="center">02/23/2018 in Kresge G2 (HSPH)</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Accessing_public_genomic_data/">Accessing public data for genomics</a></td>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Intro_shell/">Intro to shell</a></td>
<td align="center">04/27/2018 in FXB G12 (HSPH)</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Git-Github/">Version control using Git and Github</a></td>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Intro_shell/">Intro to shell</a></td>
<td align="center">05/25/2018 in Kresge 502 (HSPH)</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Exploring_variants_with_GEMINI/">Exploring genomic variants using GEMINI <em>(In development)</em></a></td>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Intro_shell/">Intro to shell</a></td>
<td align="center">TBD</td>
</tr>
<tr>
<td align="center">Introduction to tidyverse R packages for data visualization <em>(In development)</em></td>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/IntroR_ggplot2/">Intro to R</a></td>
<td align="center">TBD</td>
</tr>
</tbody>
</table>
<h3 id="in-depth-next-generation-sequencing-analysis-course">In-Depth Next Generation Sequencing Analysis Course</h3>
<p><a href="https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/">This intensive course</a> runs for between <strong>8 to 12 days</strong> and is aimed at bench biologists interested in learning how to <strong>independently</strong> perform NGS data analyses using <em>best practices</em>. Topics include:</p>
<ul>
<li>High-performance computing</li>
<li>Best practice workflows for NGS data analysis (RNA-seq, ChIP-seq, Variant calling)</li>
<li>R for statistical analysis and data visualization</li>
<li>Functional analysis with gene lists</li>
<li>Additional skils and tools for better reproducibility like reports with RMarkdown, version control with Git/Github, etc.</li>
</ul>
<h2 id="contact-us">Contact us:</h2>
<p><strong>Email:</strong> <a href="mailto:hbctraining@hsph.harvard.edu">hbctraining@hsph.harvard.edu</a></p>
<p><strong>Webpage:</strong> <a href="http://bioinformatics.sph.harvard.edu">http://bioinformatics.sph.harvard.edu</a></p>
<p><strong>Twitter:</strong> <a href="http://twitter.com/bioinfocore">@bioinfocore</a></p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href=".." class="btn btn-neutral" title="Home"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href=".." style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js"></script>
      <script src="../search/require.js"></script>
      <script src="../search/search.js"></script>

</body>
</html>
