<!DOCTYPE html>
<html lang="en">
<head>
  
  
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    
    
    
    <link rel="shortcut icon" href="../img/favicon.ico">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
    <title>Training materials - RSK's test doc</title>
    <link href="../css/bootstrap-3.3.7.min.css" rel="stylesheet">
    <link href="../css/font-awesome-4.7.0.css" rel="stylesheet">
    <link href="../css/base.css" rel="stylesheet">
    <link rel="stylesheet" href="../css/highlight.css">
    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
        <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
        <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <script src="../js/jquery-3.2.1.min.js"></script>
    <script src="../js/bootstrap-3.3.7.min.js"></script>
    <script src="../js/highlight.pack.js"></script>
    
    <base target="_top">
    <script>
      var base_url = '..';
      var is_top_frame = false;
        
        var pageToc = [
          {title: "Training program description:", url: "#_top", children: [
              {title: "Workshops", url: "#workshops" },
              {title: "In-Depth Next Generation Sequencing Analysis Course", url: "#in-depth-next-generation-sequencing-analysis-course" },
              {title: "Contact us:", url: "#contact-us" },
          ]},
        ];

    </script>
    <script src="../js/base.js"></script> 
</head>

<body>
<script>
if (is_top_frame) { $('body').addClass('wm-top-page'); }
</script>



<div class="container-fluid wm-page-content">
  <a name="_top"></a>
    

  <div class="row wm-article-nav-buttons" role="navigation" aria-label="navigation">
    
    <div class="wm-article-nav">
      <a href=".." class="btn btn-xs btn-default pull-left">
        <i class="fa fa-chevron-left" aria-hidden="true"></i>
        Previous</a><a href=".." class="btn btn-xs btn-link">
        Home
      </a>
    </div>
    
  </div>

    

    <h1 id="training-program-description">Training program description:</h1>
<p>The training team at the Harvard Chan Bioinformatics Core provides bioinformatics training through both shorter workshops and longer in-depth courses. Our current workshops and courses are designed to help biologists become comfortable with using tools to analyse high-throughput data. We are slowly beginning to expand this repertoire to include training for researchers with more advanced bioinformatics skills.</p>
<p><em>No prior NGS or command line expertise is required for our workshops or courses unless explicitly stated.</em></p>
<h3 id="workshops">Workshops</h3>
<h4 id="introduction-to-next-generation-sequencing-ngs-analysis-series">Introduction to Next-Generation Sequencing (NGS) analysis series:</h4>
<p>The goal of these workshops (2-3 days) are to enable researchers to design their NGS studies appropriately and perform preliminary data analyses.</p>
<table>
<thead>
<tr>
<th align="center">Training topic and link to lessons</th>
<th align="center">Prerequisites</th>
<th align="center">Workshop Duration</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Intro-to-R/">R and ggplot2</a></td>
<td align="center">None</td>
<td align="center">2 days</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/">RNA-seq data analysis using High-Performance Computing</a></td>
<td align="center">None</td>
<td align="center">2 - 3 days</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Intro-to-R-with-DGE/">R and Differential Gene Expression (DGE) analysis</a></td>
<td align="center">None</td>
<td align="center">3 days</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/DGE_workshop/">Differential Gene Expression (DGE) analysis</a></td>
<td align="center"><a href="https://hbctraining.github.io/Intro-to-R/">R and ggplot2</a></td>
<td align="center">1.5 days</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Intro-to-ChIPseq/">ChIP-seq using High-Performance Computing</a></td>
<td align="center">None</td>
<td align="center">3 days</td>
</tr>
<tr>
<td align="center">Identifying variants in genome/exome sequencing data</td>
<td align="center">None</td>
<td align="center"><em>In development</em></td>
</tr>
</tbody>
</table>
<h4 id="current-topics-in-bioinformatics-series">Current topics in bioinformatics series:</h4>
<p>These short workshops (half-day or less) are designed to allow researchers, who have some familiarity with R or bash, to learn new tools and methods. Only a subset of the short workshops topics are listed and linked below, for the full list please <a href="https://hbctraining.github.io/Training-modules/">click here</a>.</p>
<table>
<thead>
<tr>
<th align="center">Training topic and link to lessons</th>
<th align="center">Prerequisites</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/IntroR_ggplot2/">Introduction to R &amp; Visualizations with ggplot2</a></td>
<td align="center">None</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Visualization_in_R/">Plotting and visualization in R using ggplot2</a></td>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/IntroR_ggplot2/">Intro to R</a></td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/DGE-functional-analysis/">Functional analysis of gene lists</a></td>
<td align="center">Beginner R or <a href="https://hbctraining.github.io/Intro-to-R/">IntroR workshop</a></td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Rmarkdown/">Reproducible research using R (Rmarkdown: report generation)</a></td>
<td align="center">Beginner R or <a href="https://hbctraining.github.io/Intro-to-R/">IntroR workshop</a></td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Intro_shell/">Introduction to shell/bash</a></td>
<td align="center">None</td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Intermediate_shell/">Intermediate shell/bash</a></td>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Intro_shell/">Intro to shell</a></td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Accessing_public_genomic_data/">Accessing public data for genomics</a></td>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Intro_shell/">Intro to shell</a></td>
</tr>
<tr>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Git-Github/">Version control using Git and Github</a></td>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Intro_shell/">Intro to shell</a></td>
</tr>
<tr>
<td align="center">Exploring genomic variants using GEMINI <em>(In development)</em></td>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/Intro_shell/">Intro to shell</a></td>
</tr>
<tr>
<td align="center">Introduction to tidyverse R packages for data visualization <em>(In development)</em></td>
<td align="center"><a href="https://hbctraining.github.io/Training-modules/IntroR_ggplot2/">Intro to R</a></td>
</tr>
</tbody>
</table>
<h3 id="in-depth-next-generation-sequencing-analysis-course">In-Depth Next Generation Sequencing Analysis Course</h3>
<p><a href="https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/">This intensive course</a> runs for between <strong>8 to 12 days</strong> and is aimed at bench biologists interested in learning how to <strong>independently</strong> perform NGS data analyses using <em>best practices</em>. Topics include:</p>
<ul>
<li>High-performance computing</li>
<li>Best practice workflows for NGS data analysis (RNA-seq, ChIP-seq, Variant calling)</li>
<li>R for statistical analysis and data visualization</li>
<li>Functional analysis with gene lists</li>
<li>Additional skils and tools for better reproducibility like reports with RMarkdown, version control with Git/Github, etc.</li>
</ul>
<h2 id="contact-us">Contact us:</h2>
<p><strong>Email:</strong> <a href="mailto:hbctraining@hsph.harvard.edu">hbctraining@hsph.harvard.edu</a></p>
<p><strong>Webpage:</strong> <a href="http://bioinformatics.sph.harvard.edu">http://bioinformatics.sph.harvard.edu</a></p>
<p><strong>Twitter:</strong> <a href="http://twitter.com/bioinfocore">@bioinfocore</a></p>

  <br>
    

  <div class="row wm-article-nav-buttons" role="navigation" aria-label="navigation">
    
    <div class="wm-article-nav">
      <a href=".." class="btn btn-xs btn-default pull-left">
        <i class="fa fa-chevron-left" aria-hidden="true"></i>
        Previous</a><a href=".." class="btn btn-xs btn-link">
        Home
      </a>
    </div>
    
  </div>

    <br>
</div>

<footer class="col-md-12 wm-page-content">
  <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a> using <a href="https://github.com/gristlabs/mkdocs-windmill">Windmill</a> theme by Grist Labs.</p>
</footer>

</body>
</html>