# Training program description:

The training team at the Harvard Chan Bioinformatics Core provides bioinformatics training through both shorter workshops and longer in-depth courses. Our current workshops and courses are designed to help biologists become comfortable with using tools to analyse high-throughput data. We are slowly beginning to expand this repertoire to include training for researchers with more advanced bioinformatics skills.

*No prior NGS or command line expertise is required for our workshops or courses unless explicitly stated.*

### Workshops

#### Introduction to Next-Generation Sequencing (NGS) analysis series: 

The goal of these workshops (2-3 days) are to enable researchers to design their NGS studies appropriately and perform preliminary data analyses.

| Training topic and link to lessons | Prerequisites | Workshop Duration |
:----------:|:----------:|:----------:|
| [R and ggplot2](https://hbctraining.github.io/Intro-to-R/) | None | 2 days |
| [RNA-seq data analysis using High-Performance Computing](https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/) | None | 2 - 3 days |
| [R and Differential Gene Expression (DGE) analysis](https://hbctraining.github.io/Intro-to-R-with-DGE/) | None | 3 days |
| [Differential Gene Expression (DGE) analysis](https://hbctraining.github.io/DGE_workshop/) | [R and ggplot2](https://hbctraining.github.io/Intro-to-R/) | 1.5 days |
| [ChIP-seq using High-Performance Computing](https://hbctraining.github.io/Intro-to-ChIPseq/) | None  | 3 days |
| Identifying variants in genome/exome sequencing data  | None | *In development* |

#### Current topics in bioinformatics series:

These short workshops (half-day or less) are designed to allow researchers, who have some familiarity with R or bash, to learn new tools and methods. Only a subset of the short workshops topics are listed and linked below, for the full list please [click here](https://hbctraining.github.io/Training-modules/).

| Training topic and link to lessons | Prerequisites | 
:----------:|:----------:|
| [Introduction to R & Visualizations with ggplot2](https://hbctraining.github.io/Training-modules/IntroR_ggplot2/) | None |
| [Plotting and visualization in R using ggplot2](https://hbctraining.github.io/Training-modules/Visualization_in_R/) | [Intro to R](https://hbctraining.github.io/Training-modules/IntroR_ggplot2/) |
| [Functional analysis of gene lists](https://hbctraining.github.io/Training-modules/DGE-functional-analysis/) | Beginner R or [IntroR workshop](https://hbctraining.github.io/Intro-to-R/) |
| [Reproducible research using R (Rmarkdown: report generation)](https://hbctraining.github.io/Training-modules/Rmarkdown/) | Beginner R or [IntroR workshop](https://hbctraining.github.io/Intro-to-R/) |
| [Introduction to shell/bash](https://hbctraining.github.io/Training-modules/Intro_shell/) | None | 
| [Intermediate shell/bash](https://hbctraining.github.io/Training-modules/Intermediate_shell/) | [Intro to shell](https://hbctraining.github.io/Training-modules/Intro_shell/) | 
| [Accessing public data for genomics](https://hbctraining.github.io/Training-modules/Accessing_public_genomic_data/) | [Intro to shell](https://hbctraining.github.io/Training-modules/Intro_shell/) | 
| [Version control using Git and Github](https://hbctraining.github.io/Training-modules/Git-Github/) | [Intro to shell](https://hbctraining.github.io/Training-modules/Intro_shell/) |
| Exploring genomic variants using GEMINI *(In development)* | [Intro to shell](https://hbctraining.github.io/Training-modules/Intro_shell/) |
| Introduction to tidyverse R packages for data visualization *(In development)* | [Intro to R](https://hbctraining.github.io/Training-modules/IntroR_ggplot2/) | 

### In-Depth Next Generation Sequencing Analysis Course

[This intensive course](https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/) runs for between **8 to 12 days** and is aimed at bench biologists interested in learning how to **independently** perform NGS data analyses using *best practices*. Topics include:

  * High-performance computing
  * Best practice workflows for NGS data analysis (RNA-seq, ChIP-seq, Variant calling)
  * R for statistical analysis and data visualization
  * Functional analysis with gene lists
  * Additional skils and tools for better reproducibility like reports with RMarkdown, version control with Git/Github, etc.

## Contact us:

**Email:** [hbctraining@hsph.harvard.edu](mailto:hbctraining@hsph.harvard.edu)

**Webpage:** [http://bioinformatics.sph.harvard.edu](http://bioinformatics.sph.harvard.edu)

**Twitter:** [@bioinfocore](http://twitter.com/bioinfocore)
